samples/tool/SFSClientTool.cpp

```diff
@@ -309,7 +301,46 @@ void DisplayResults(const std::vector<AppContent>& contents)
         out.push_back(j);
     }
 
-    PrintLog(out.dump(2 /*indent*/));
+    return out;
+}
+
+void DisplayResults(const json& results)
+{
+    if (results.empty())
+    {
+        std::cout << "No results found." << std::endl;
+        return;
+    }
+
+    PrintLog("Content found:");
+    PrintLog(results.dump(2 /*indent*/));
+}
+
+Result JSONToFile(const json& results, const std::string& filename)
+{
+    const std::filesystem::path filepath = std::filesystem::absolute(filename);

```

Comment:
check results.empty first, or do you intentionally want to create an empty file if no results found?